pkg/storage/utils/eosfs/eosfs.go

```diff
@@ -0,0 +1,1341 @@
+// Copyright 2018-2020 CERN
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// In applying this license, CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+package eosfs
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/url"
+	"os"
+	gouser "os/user"
+	"path"
+	"regexp"
+	"strconv"
+	"strings"
+
+	"github.com/cs3org/reva/pkg/appctx"
+	"github.com/cs3org/reva/pkg/eosclient"
+	"github.com/cs3org/reva/pkg/mime"
+	"github.com/cs3org/reva/pkg/storage"
+	"github.com/cs3org/reva/pkg/storage/utils/acl"
+	"github.com/cs3org/reva/pkg/storage/utils/grants"
+	"github.com/cs3org/reva/pkg/storage/utils/templates"
+	"github.com/cs3org/reva/pkg/user"
+	"github.com/pkg/errors"
+
+	userpb "github.com/cs3org/go-cs3apis/cs3/identity/user/v1beta1"
+	provider "github.com/cs3org/go-cs3apis/cs3/storage/provider/v1beta1"
+	types "github.com/cs3org/go-cs3apis/cs3/types/v1beta1"
+	"github.com/cs3org/reva/pkg/errtypes"
+)
+
+const (
+	refTargetAttrKey = "reva.target"
+)
+
+var hiddenReg = regexp.MustCompile(`\.sys\..#.`)
+
+// Config holds the configuration details for the EOS fs.
+type Config struct {
```

Comment:
Can this be somehow re-used?

Reply:
Yes, we can use only a single config in utils/eosfs/eosfs.go, but I wanted to keep it separately in eos and eoshome so that it can be picked up by cato. Since both of them have a few differences in their configs, using a common config will not be reflected in the documentation.

Reply:
Okay